storage/storage-interface/src/finality_view.rs

```diff
@@ -0,0 +1,159 @@
+        fin_ledger_info
+            .as_ref()
+            .map(|li| li.ledger_info().version())
+            .ok_or_else(|| AptosDbError::NotFound("finalized version".into()))
+    }
+
+    fn get_latest_state_checkpoint_version(&self) -> Result<Option<Version>> {
+        let fin_ledger_info = self.finalized_ledger_info.read().unwrap();
+        let version = fin_ledger_info
+            .as_ref()
+            .map(|li| li.ledger_info().version());
+        Ok(version)
+    }
+
+    // TODO: override any other methods needed to maintain the illusion.
+}
+
+#[cfg(test)]
+mod tests {
+    use aptos_types::{aggregate_signature::AggregateSignature, ledger_info::LedgerInfo};
+
+    use super::*;
+    use crate::mock::MockDbReaderWriter;
+
+    #[test]
+    fn test_get_latest_ledger_info() {
+        // If the mock is changed to be stateful, this should be ref-counted
+        // and shared with the view.
+        let mock = MockDbReaderWriter;
+        let view = FinalityView::new(MockDbReaderWriter);
+
+        let ledger_info = view.get_latest_ledger_info_option().unwrap();
+        assert_eq!(ledger_info, None);
+        let blockheight = 1;
+
+        // Set the finalized ledger info
+        view.set_finalized_block_height(blockheight).unwrap();
+
+        // Capture the block event once
+        let (_start_ver, end_ver, block_event) =
+            mock.get_block_info_by_height(blockheight).unwrap();
+        let block_hash = block_event.hash().unwrap(); // Used to verify hash is generated
+
+        let block_info = BlockInfo::new(
+            block_event.epoch(),
+            block_event.round(),
+            block_hash,
+            HashValue::zero(),
+            end_ver,
+            block_event.proposed_time(),
+            None,
+        );
+        let ledger_info = LedgerInfo::new(block_info, HashValue::zero());
+        let expected_ledger_info =
+            LedgerInfoWithSignatures::new(ledger_info, AggregateSignature::empty());
+
+        // Get the latest ledger info after setting it
+        let ledger_info = view.get_latest_ledger_info_option().unwrap().unwrap();
+
+        assert_eq!(ledger_info, expected_ledger_info);
+    }
+
+    #[test]
+    fn test_get_latest_version() {
+        let view = FinalityView::new(MockDbReaderWriter);
+        let res = view.get_latest_version();
+        assert!(res.is_err());
+        let blockheight = 1;
+        view.set_finalized_block_height(blockheight).unwrap();
+        let version = view.get_latest_version().unwrap();
+        assert_eq!(version, 1);
+    }
+
+    #[test]
+    fn test_get_latest_state_checkpoint_version() {
+        let view = FinalityView::new(MockDbReaderWriter);
+        let version = view.get_latest_state_checkpoint_version().unwrap();
+        assert_eq!(version, None);
+        view.set_finalized_block_height(1).unwrap();
+        let version = view.get_latest_state_checkpoint_version().unwrap();
```

Comment:
I may be a bit behind the ball here on our latest means of handling things.  But...

Is there a reason we shouldn't have a test here that actually checks whether the state made available corresponds to what we would expect for a given ledger version/block height? 

I suppose since it seems we are still always working off of checkpoints this is effectively checked by the original `aptos-core` tests.  But, it would be good to perhaps link to where those are.

If we change this in the future, should we include checks at this level that would make sure this fin view actually marries up with the proper `StateView`?

Reply:
I'll add a test specifically for a state view created around `FinalityView` to check that the states are only available to the finality version.

Reply:
The relevant view is the one obtained by the `Context::latest_state_view*` methods.
I wrote a stub of a test for it, but proper testing would need more support from the mock.

Reply:
The test in https://github.com/movementlabsxyz/movement/pull/91/commits/3541365d18aa36102464b7d1dd8079870f9f2b54 tests that it works on a real populated database.